src/Element.js

```diff
@@ -0,0 +1,341 @@
+}
+
+/**
+ * @param {*} element
+ * @return {object}
+ */
+export function module (element) {
+	return from(element && typeof element === 'object' && element.default || element, Constant.key)
+}
+
+/**
+ * @param {*} children
+ * @param {*} target
+ * @param {*} key
+ * @return {object}
+ */
+export function portal (children, target, key) {
+	return create(target, {key: key}, children)
+}
+
+/**
+ * @param {number} type
+ * @param {Array<object>} children
+ * @param {number} key
+ * @return {object}
+ */
+export function iterable (type, children, key) {
+	return new struct(type, '#iterable', Constant.object, children, key)
+}
+
+
+/**
+ * @param {(number|string)} children
+ * @param {*} key
+ * @return {object}
+ */
+export function comment (children, key) {
+	return new struct(Constant.comment, '#comment', Constant.object, children, key)
+}
+
+/**
+ * @param {(number|string)} children
+ * @param {number} index
+ * @return {object}
+ */
+export function text (children, index) {
+	return new struct(Constant.text, '#text', Constant.object, children, Utility.hash(index))
+}
+
+/**
+ * @param {number} index
+ * @return {object}
+ */
+export function empty (index) {
+	return new struct(Constant.empty, '#empty', Constant.object, '', Utility.hash(index))
+}
+
+/**
+ * @param {*} a
+ * @param {(object|*)?} b
+ * @param {...*?}
+ * @return {object}
+ * @public
+ */
+export function create (a, b) {
+	var index = 0
+	var length = arguments.length
+	var i = properties(b) ? 2 : 1
+	var size = length - i
+	var uuid = identity(a)
+	var type = a
+	var props = i === 2 ? b : {}
+	var children = []
+	var key = props.key || Constant.key
+	var element = new struct(uuid, type, props, children, key)
+
+	if (uuid === Constant.component) {
+		if (size) {
+			for (props.children = size === 1 ? arguments[i++] : children = []; i < length; ++i) {
+				children.push(arguments[i])
+			}
+		}
+		if (type[Constant.defaultProps]) {
+			defaults(element, type[Constant.defaultProps])
+		}
+		if ("production" === 'development') {
+			Check.types(element, props, Constant.propTypes)
+		}
+	} else {
+		if (size) {
+			for (; i < length; ++i) {
+				children[index] = from(arguments[i], index++)
+			}
+		} else if (Utility.has(props, 'children')) {
+			children[index] = from(arguments[i], index)
+		}
+		if (uuid < Constant.node) {
+			return iterable(Constant.fragment, [empty(Constant.key), element, empty(Constant.key)], key)
+		}
+	}
+
+	return element
+}
+
+/**
+ * @param {*} value
+ * @param {number} index
+ * @return {object}
+ */
+export function from (value, index) {
+	if (value != null) {
+		if (value.constructor === Constant.element) {
+			return value
+		} else {
+			switch (typeof value) {
+				case 'number':
+				case 'string':
+					return text(value, index)
+				case 'function':
+				case 'object':
+					if (Utility.iterable(value)) {
+						if (index === Constant.key) {
+							return iterable(Constant.fragment, value.map(from), Utility.hash(index))
```

Comment:
While both are fragment types the difference between Constant.fragment and Constant.iterable is that `Constant.fragment` can be rendered offscreen when required. Constant.iterable is mostly used for arrays to namespace arrays to prevent key collision among siblings. i.e in `</>{[1, 2, 3], [1, 2, 3]}</>` both arrays are reconciled as separate compounds.